apple/MarkdownTextInputDecoratorShadowNode.mm

```diff
@@ -0,0 +1,255 @@
+}
+
+void MarkdownTextInputDecoratorShadowNode::overwriteMeasureContent() {
+  const auto &children = getChildren();
+  if (children.empty()) {
+    return;
+  }
+  react_native_assert(
+      children.size() == 1 &&
+      "MarkdownTextInputDecoratorView received more than one child");
+
+  const auto child = std::dynamic_pointer_cast<const TextInputShadowNode>(children[0]);
+  react_native_assert(
+      child != nullptr &&
+      "MarkdownTextInputDecoratorView received child other than a TextInput");
+
+  // don't mind this :)
+  const auto &nodeWithAccessibleYogaNode =
+      std::reinterpret_pointer_cast<const MarkdownTextInputDecoratorShadowNode>(child);
+  
+  // decorator node cannot have a measure function since it's not a leaf node
+  // but we can redirect measuring of the child input to call measureContent
+  // on the decorator
+  YGNodeSetMeasureFunc(&nodeWithAccessibleYogaNode->yogaNode_,
+                       yogaNodeMeasureCallbackConnector);
+}
+
+void MarkdownTextInputDecoratorShadowNode::appendChild(
+    const ShadowNode::Shared &child) {
+  YogaLayoutableShadowNode::appendChild(child);
+
+  overwriteMeasureContent();
+}
+
+void MarkdownTextInputDecoratorShadowNode::replaceChild(
+    const ShadowNode &oldChild, const ShadowNode::Shared &newChild,
+    size_t suggestedIndex) {
+  YogaLayoutableShadowNode::replaceChild(oldChild, newChild, suggestedIndex);
+
+  overwriteMeasureContent();
+};
+
+Size MarkdownTextInputDecoratorShadowNode::measureContent(
+    const LayoutContext &layoutContext,
+    const LayoutConstraints &layoutConstraints) const {
+  const auto &children = getChildren();
+  react_native_assert(
+      children.size() == 1 &&
+      "MarkdownTextInputDecoratorView received wrong number of children");
+
+  const auto child =
+      std::static_pointer_cast<const TextInputShadowNode>(children.at(0));
+
+  child->ensureUnsealed();
+
+  // apply markdown formatting before measuring the child
+  const auto &mutableChild =
+      std::const_pointer_cast<TextInputShadowNode>(child);
+  applyMarkdownStylesToTextInputState(mutableChild, layoutContext);
+
+  return child->measureContent(layoutContext, layoutConstraints);
+}
+
+void MarkdownTextInputDecoratorShadowNode::layout(LayoutContext layoutContext) {
+  YogaLayoutableShadowNode::layout(layoutContext);
+
+  const auto &children = getChildren();
+  react_native_assert(
+      children.size() == 1 &&
+      "MarkdownTextInputDecoratorView didn't receive exactly one child");
+
+  const auto child =
+      std::static_pointer_cast<const TextInputShadowNode>(children.at(0));
+
+  child->ensureUnsealed();
+
+  const auto &mutableChild =
+      std::const_pointer_cast<TextInputShadowNode>(child);
+
+  // TODO: this may not be the correct way to do this
+  auto childMetrics = child->getLayoutMetrics();
+  setLayoutMetrics(childMetrics);
+
+  childMetrics.frame.origin = Point{};
+  mutableChild->setLayoutMetrics(childMetrics);
```

Comment:
Let's add comments as well

Reply:
Is that ok? [7250899](https://github.com/Expensify/react-native-live-markdown/pull/566/commits/7250899ab32fa4aa7481befeaa20e5be46cc8388)